Mage.Sets/src/mage/cards/f/ForgehammerCenturion.java

```diff
@@ -0,0 +1,55 @@
+package mage.cards.f;
+
+import mage.MageInt;
+import mage.abilities.common.AttacksTriggeredAbility;
+import mage.abilities.common.PutIntoGraveFromBattlefieldAllTriggeredAbility;
+import mage.abilities.costs.common.RemoveCountersSourceCost;
+import mage.abilities.effects.common.DoIfCostPaid;
+import mage.abilities.effects.common.combat.CantBlockTargetEffect;
+import mage.abilities.effects.common.counter.AddCountersSourceEffect;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Duration;
+import mage.constants.SubType;
+import mage.counters.CounterType;
+import mage.filter.StaticFilters;
+import mage.target.common.TargetCreaturePermanent;
+
+import java.util.UUID;
+
+public class ForgehammerCenturion extends CardImpl {
+    public ForgehammerCenturion(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{R}");
+        this.addSubType(SubType.PHYREXIAN);
+        this.addSubType(SubType.WARRIOR);
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(2);
+
+        //Whenever another creature or artifact you control is put into a graveyard from the battlefield, put an oil
+        //counter on Forgehammer Centurion.
+        this.addAbility(new PutIntoGraveFromBattlefieldAllTriggeredAbility(
+                new AddCountersSourceEffect(CounterType.OIL.createInstance()),
+                false, StaticFilters.FILTER_CONTROLLED_ANOTHER_ARTIFACT_OR_CREATURE, false
+        ));
+
+        //Whenever Forgehammer Centurion attacks, you may remove two oil counters from it. When you do, target creature
+        //can’t block this turn.
+        AttacksTriggeredAbility attacksTriggeredAbility = new AttacksTriggeredAbility(new DoIfCostPaid(
```

Comment:
This triggered ability must trigger in two halves, with the option to remove two oil counters from this being presented to the controller, *then* allowing them to select a target for the "can't be blocked" half if they choose to pay the cost. See [Ambulatory Edifice](https://github.com/magefree/mage/blob/7b1c703d21cce54d2fb33f31400c1b7aee8fb208/Mage.Sets/src/mage/cards/a/AmbulatoryEdifice.java) or any of these [other "when you do" cards](https://scryfall.com/search?q=o%3A%22when+you+do%22+-o%3A%22exert%22+not%3Adigital+not%3Afunny&unique=cards&as=grid&order=name)' implementation for examples

Reply:
done